### ISSUE TITLE
Enhancement: Add `Sentry` integration for 2.0

### ISSUE THREAD

**Issue by JacobCoffee** [litestar-org/litestar#1514]
### Summary

We have a [sentry](https://docs.sentry.io/) integration for 1.x. We need to update that integration, but we did not write any docs.

- https://docs.sentry.io/platforms/python/ should have a guide about Litestar usage.

### Basic Example

Please follow the [Sentry Integration Guide](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md#adding-a-new-integration-checklist) with special emphasis on the docs

Per the [Sentry Integration Guide](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md#adding-a-new-integration-checklist): 

#### Write the [docs](https://github.com/getsentry/sentry-docs). Answer the following questions:

- What does your integration do? Split in two sections: Executive summary at top and exact behavior further down.
- Which version of the SDK supports which versions of the modules it hooks into?
- One code example with basic setup.
- Make sure to add integration page to python/index.md (people forget to do that all the time). 

### Drawbacks and Impact

N/A

### Unresolved questions

1. How do we handle the renaming of the current integration per Sentry guidelines?

<a href="https://polar.sh/litestar-org/litestar/issues/1514">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1514/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1514/pledge.svg">
</picture>
</a>

**Comment by Goldziher** [comment]
I dont think this is an issue for this repository. Sentry is a for-profit company, and we should not contribute to them out of our own charity. Please open an issue on the sentry side of the fence, and they can take it from there - or not.

**Comment by janluke** [comment]
@Goldziher Does the current integration for Starlite need major modifications or it's just a matter of renaming `starlite` with `litestar`?

**Comment by janluke** [comment]
I asked the question because we need this (at the company I work for) and we're considering contributing, I just wanted to have an idea on the effort needed, considering that we're new to Litestar and never contributed to Sentry. Thanks.

**Comment by peterschutt** [comment]
I'm a heavy user of sentry also, so would like to see this done. I'd have to look at the code more thoroughly to comment, but I'd imagine there will be some fairly significant changes required as starlite -> litestar was a bit of a jump.

**Comment by JacobCoffee** [comment]
It would be welcome and we could help with the transition when needed.

**Comment by bpereto** [comment]
I tried to adapt the implementation mentioned in https://github.com/getsentry/sentry-python/blob/f570a9966252920bdb221101d596eb029497b0e9/sentry_sdk/integrations/starlite.py#L51

https://gist.github.com/bpereto/8b49893f4a19740c19bf336af94aa95b#file-sentry-py

The basic works, but I struggled with the `retrieve_user_from_scope`.

**Comment by JacobCoffee** [comment]
@kedod you might work with @guacs as well. There is a thread in the [development discord channel](https://discord.com/channels/919193495116337154/1183945128637898812). 

Happy to have anyone take this if they have the capacity!

**Comment by bpereto** [comment]
> @bpereto Do you need a hand with this one?

I'm not sure how the internals of litestar work, but my implementation attempt seems do it's job.

maybe you know the framework better and how to implement it correctly?

Im unsure if the references are the newest versions.

**Comment by guacs** [comment]
@kedod if you want to take this, please feel free :)

**Comment by peterschutt** [comment]
The logging integration gives you _something_, but obviously not ideal as no spans etc.

I spoke with one of their engineers at pycon, and he mentioned they might be exposing some new apis that would help us to build an integration without monkey-patching the application (which would be my preference). I'll be watching that space, but as a sentry user myself, this is a problem I'd like to see solved too.

**Comment by rafalkrupinski** [comment]
OK, looks like I forgot to configure DSN :D
ASGI integration actually does something.

If one was to contribute to the integration:
- did anyone started a sentry fork to put the code in?
- is there a guide for integrating with python servers or a reference integration?

I like Litestar code, but I know nothing about sentry internals.

**Comment by JacobCoffee** [comment]
This is done in https://github.com/getsentry/sentry-python/pull/3358
